pkg/clustermesh/common/remote_cluster.go

```diff
@@ -139,11 +140,21 @@ func (rc *remoteCluster) restartRemoteConnection() {
 			DoFunc: func(ctx context.Context) error {
 				rc.releaseOldConnection()
 
-				extraOpts := rc.makeExtraOpts()
+				clusterLock := newClusterLock()
+
+				extraOpts := rc.makeExtraOpts(clusterLock)
 
 				backend, errChan := kvstore.NewClient(ctx, kvstore.EtcdBackendName,
 					rc.makeEtcdOpts(), &extraOpts)
 
+				ctx, cancel := context.WithCancel(ctx)
+				rc.wg.Add(1)
```

Comment:
assuming rc.wg.Add() is newly added, would expect rc.wg.Wait() to be added.
Otherwise, the newly added thread could be leaking.

Reply:
Hi @ccfish2,

Thanks for your review. This commit just moved that logic a few lines earlier, and `wg.Wait()` was already performed in [`releaseOldConnection()`](https://github.com/cilium/cilium/blob/4f8a3a627ebdc5c2e4733256350ffb73e2d4bae0/pkg/clustermesh/common/remote_cluster.go#L117).